### ISSUE TITLE
SAG Mill Eating Secondary Output

### ISSUE THREAD

**Issue by magerooster** [SleepyTrousers/EnderIO#2345]
Forge Version: 10.13.2.1343
EnderIO Version: 1.7.10-2.2.8.367

I noticed when I had a SAG mill directly outputting to an alloy furnace that while the alloy furnace would pull the primary output (in this case, the pulverized iron from the iron ore), it would (properly) not pull the others, as it wasn't keeping up.

However, once the pulverized nickel stack filled up, the SAG mill (improperly) kept running, producing more pulverized iron and (I assume) eating the excess nickel being made at that point. I expected the SAG mill to stop until it had space for the secondary outputs. And as the three secondary slots all had something in them (nickel, tin, cobble), it had nowhere else to put it.

tl;dr: SAG mill only checks space for primarily output before performing an operation and loses the when the secondary outputs are full.

**Comment by MatthiasMann** [comment]
Actually the SAG Mill asks the recipe for it's outputs - and a random number is chosen. If the outputs fit the process is started - otherwise it waits 1 second.

I tried to fix that - but it is not as easy as I expected. I might give it another go.

**Comment by braddr** [comment]
'eating' the secondary output is a consequence of that algorithm.  It will keep trying until the no secondary output option its and run it.  The only way to NOT do that is to block the run when the outputs will not fit before the chance portion is done.  Or declare the current behavior expected.

**Comment by magerooster** [comment]
Perhaps not running unless the maximum output is available?

**Comment by tterrag1098** [comment]
Yeahh I agree with @magerooster, it should not run if exceeding the storage space available is possible *at all*, the current algorithm is unfair.

**Comment by MatthiasMann** [comment]
I'll try to chance it as @braddr suggested - if that is doable without rewriting everything.
The other idea I had was to freeze the random number generation - so that it won't re-roll for each check unless the check has passed.

**Comment by braddr** [comment]
I probably shouldn't try reading code at this hour, but.. I _think_ it might just work to change the 'chance' variable in this line to whatever value means always succeed:

https://github.com/SleepyTrousers/EnderIO/blob/master/src/main/java/crazypants/enderio/machine/AbstractPoweredTaskEntity.java#L276
